test/@types/vitest.d.ts

```diff
@@ -238,6 +268,6 @@ interface PokemonMatchers {
    * If the Pokemon's moveset has been set via {@linkcode Overrides.MOVESET_OVERRIDE}/{@linkcode Overrides.ENEMY_MOVESET_OVERRIDE}
    * or does not contain exactly one copy of `moveId`, this will fail the test.
    */
-  toHaveUsedPP(moveId: MoveId, ppUsed: number | "all"): void;
+  toHaveUsedPP<P extends number | "all">(moveId: MoveId, ppUsed: If<IsNumericLiteral<P>, Integer<P>, P>): void;
```

Comment:
```suggestion
  toHaveUsedPP<P extends number | "all">(moveId: MoveId, ppUsed: If<IsNumericLiteral<P>, NonNegativeInteger<P>, P>): void;
```
Pretty sure a move should never use negative PP.